packages/grafana-e2e/src/support/scenario.ts

```diff
@@ -35,13 +35,13 @@ export const e2eScenario = ({
 
       afterEach(() => {
         getScenarioContext().then(({ lastAddedDashboardUid, lastAddedDataSource }: any) => {
-          if (lastAddedDataSource) {
-            Flows.deleteDataSource(lastAddedDataSource);
-          }
-
           if (lastAddedDashboardUid) {
             Flows.deleteDashboard(lastAddedDashboardUid);
           }
+
+          if (lastAddedDataSource) {
```

Comment:
does the order change here really matter?

Reply:
Yes, it makes tests not fail. It's desribed in the commit messages